src/components/WeeklySummary/WeeklySummary.jsx

```diff
@@ -0,0 +1,348 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Alert, Container, Row, Col, Form, FormGroup, Label, Input,
+  CustomInput, Button, TabContent, TabPane, Nav, NavItem, NavLink,
+} from 'reactstrap';
+import './WeeklySummary.css';
+import { connect } from 'react-redux';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons';
+import { Editor } from '@tinymce/tinymce-react';
+import { getWeeklySummaries, updateWeeklySummaries } from '../../actions/weeklySummaries'
+import DueDateTime from './DueDateTime';
+import moment from 'moment';
+import 'moment-timezone';
+import Loading from '../common/Loading';
+import Joi from 'joi';
+import { toast } from "react-toastify";
+import { WeeklySummaryContentTooltip, MediaURLTooltip } from './WeeklySummaryTooltips';
+import classnames from 'classnames';
+
+// Need this export here in order for automated testing to work.
+export class WeeklySummary extends Component {
+
+  state = {
+    formElements: {
+      summary: '',
+      summaryLastWeek: '',
+      summaryBeforeLast: '',
+      mediaUrl: '',
+      weeklySummariesCount: 0,
+      mediaConfirm: false
+    },
+    dueDate: moment().tz('America/Los_Angeles').endOf('week').toISOString(),
+    dueDateLastWeek: moment().tz('America/Los_Angeles').endOf('week').subtract(1, 'week').toISOString(),
+    dueDateBeforeLast: moment().tz('America/Los_Angeles').endOf('week').subtract(2, 'week').toISOString(),
```

Comment:
.tz('America/Los_Angeles') 
Should these be set? Or should this be referenced from a global const and passed down?

Reply:
Good question, yes I need them set as fallback to cover the case when for whatever reason the dueDates are not coming from the database. You can see when I use them in `componentDidMount()`. There are various scenarios that this could happen, so without that bit the component might not work properly, it needs the due dates, so in those rare cases pull local time. Think of it as fallback. In general local time obtained through moment.js is taken from the local machine which makes it vary unreliable, you could set your computer's date to 5 yrs in the past or in the future even though the odds of this happening in real life are low it could happen. So in the component I make sure to always pull those dates from the server **first**, this is absolutely must but if for whatever reason those dates aren't there I pull local time, as simple as that.